src/abstract-string-valued.store.ts

```diff
@@ -0,0 +1,48 @@
+import { IAsyncStore } from '@eppo/js-client-sdk-common';
+
+export abstract class AbstractStringValuedAsyncStore<T> implements IAsyncStore<T> {
```

Comment:
After adding the concept of an an "age" to the local-storage backed store, it was _very_ similar to the chrome storage backed store. So I consolidated all the logic into a shared parent abstract class.

Reply:
I like this idea; thanks for putting together an example, too! The old adage is to favor composition over inheritance!

I think I went down this path because of the "is a" relationship. Chrome Storage is a String-valued store. Local Storage is a String-valued store. But it can also be interpreted as "has a"-- A string-valued store has a chrome storage engine, or it has a local storage engine. (Whereas an animal cannot have a dog). Since that second relationship _also_ makes sense, I say the tie goes to composition!

Reply:
would recommend leaving top level class comments, this looks like a good candidate!